### ISSUE TITLE
Multil Template Fitter planned?

### ISSUE THREAD

**Issue by Innerwolf** [bat/BAT.jl#394]
Hey, I wanted to ask if you planned on implementing the BCMultiTemplateFitter from the C++ version. Or if there is anything similar in the Julia version. This would be very useful when fitting multiple spectral templates from simulation to experimental reference data.

**Comment by oschulz** [comment]
Yes, we'd like to offer a few "standard models", like BAT-C++ did. Since we can be more modular in Julia, this wouldn't necessarily have to be part of BAT.jl itself, we can do this in separate packages.

@Innerwolf on what time scale do you need this?

**Comment by Innerwolf** [comment]
Hey @oschulz, thanks for the quick reply!

**Comment by Innerwolf** [comment]
Hey @oschulz! Thanks for the quick reply. Great news. I wouldn't mind if you move this to another Julia package.

My timescale is not so urgent, I'm currently developing a program based on the BCMTF model from C++ and it works rather well. Right now I don't see an immediate reason or benefit from switching to the Julia version. But going into the future I might want to switch to the actively maintained version of BAT.

I would have one more question: Will using HMC be possible when using such a multi-template fitter? Or am I stuck with MHA anyway?

**Comment by oschulz** [comment]
> Will using HMC be possible when using such a multi-template fitter?

Yes, absolutely (since it'll be auto-differentiable). All samplers, including nested sampling, will work with it.

**Comment by Innerwolf** [comment]
Okay I see, thanks. Feel free to resolve this issue if needed!

**Comment by oschulz** [comment]
Let's keep it open until we actually have it implemented somewhere.

**Comment by oschulz** [comment]
Update on this: A Julia implementation with BAT.jl interface for the [upcoming HS3 standard](https://www.overleaf.com/project/63da374d94174e292e0fe9d9) is currently [under development](https://indico.cern.ch/event/1264371/contributions/5338176/attachments/2620029/4529625/IESC%20-%20HS3%20-%20A%20serialization%20standard%20for%20statistical%20models%20in%20high%20energy%20physics.pdf). HS3 covers template fits and much more..
